geometry.go

```diff
@@ -140,3 +140,9 @@ func (s Size) Subtract(v Vector2) Size {
 func (s Size) SubtractWidthHeight(width, height float32) Size {
 	return Size{s.Width - width, s.Height - height}
 }
+
+// RectangleRadius describes a radius in pixel or in percentage
+type Radius struct {
+	Pixel   float32
```

Comment:
I don't think this is a good idea - the toolkit API is all vector and therefore pixel parameters should not be used.

Can we just use `float32` instead of `Radius` - the `Width` and `Height` are the same so that will be what developers are expecting. We also don't have percentage based calculations elsewhere so I don't think it's needed.

Reply:
from @andydotxyz in https://github.com/fyne-io/fyne/pull/3121.
> It may be worth noting that if Radius == Height/2 then the ends of the rectangle should be a semi-circle, but I don't think it is, so when adjusting the scaling calculations maybe check this too?

This statement was for me the wish to have a percentage alternative for radius. 

How to solve this? The developer uses a container that changes the size depending on the windows' size.  The included rectangle changes the size dynamically too.
- The rectangle should look always the same, eg. semi-circle

In this case it would even be necessary to define the StrokeWidth as percentage to get an identical looking rectangle.
If you don't like the struct it could be named:
```go
Radius float32
RadiusRelation  float32 //RadiusDynamic RadiusPercent
StrokeWidth float32
StrokeWidthRelation float32 // StrokeWidthDynamic StrokeWidthPercent
```
It could be possible to remove the percentage from the rectangle and to define a separate shape eg ```FlexRectagle```, ```DynamicRectagle```, where the Radius and SrokeWidth are always handled in percentage. This could be implemented later after this PR. I would prefer the solution with an own Shape-Type. It would be faster and would have less complexity.

Please let me know how to change the code.

Reply:
I think you are putting to much smart into dumb canvas objects.
Something resizing the rectangle can set it's radius to be different if it is trying to maintain a specific shape.
Just like if someone wants a square - they have to ensure that the width and height match even if the container space is rectangular not square.

I don't think it's worth a more complicated API at this level to handle corner cases like that automatically. The canvasobjects are our primitive draw type - widgets using them are where the layout and render details are calculated.

Reply:
OK, I will remove the percentage and the Radius will be ```Radius float```.

Reply:
Refactor Radius done.